### PR TITLE
Fixes for SonarCloud

### PR DESCRIPTION
**Pull request by Danielius1922**

### CODE REVIEW

File: message/pool/message_test.go [hunk 1]
```diff
@@ -140,3 +141,122 @@ func TestMessageMustSetPath(t *testing.T) {
...
+				value: []byte(strings.Repeat("a", maxETagPathLen+1)),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			msg := pool.NewMessage()
+			msg.SetETag(tt.args.value)
+
+			value, err := msg.ETag()
```
Comment (Danielius1922) [inline message/pool/message_test.go:253]: @jkralik we have both `GetETag` and `ETag` functions, I think that's by mistake right?

Also how strict should we take the option definition? It defines min length as 1 and max length as 8, but the setter doesn't have problem with values outside of this range.

Comment (jkralik) [reply message/pool/message_test.go:253]: pls remove `GetETag`

I think we can be strict.

Comment (Danielius1922) [reply message/pool/message_test.go]: `GetETag` removed.

I've added bounds check to `SetETag`, though technically that changes public API. But IMO adding `error` return value shouldn't be problematic.

File: message/pool/message.go [hunk 2]
```diff
@@ -135,11 +135,20 @@ func (r *Message) SetCode(code codes.Code) {
 	r.isModified = true
 }
 
-func (r *Message) SetETag(value []byte) {
+// SetETag insert/update ETag option.
+//
+// Option definition:
+// 	- format: opaque, length: 1-8, repeatable
```
Comment (Danielius1922) [inline message/pool/message.go:141]: @jkralik apparently, ETag should be repeatable. I guess we should add a `AddETag` function and the setter should remove all existing ETag options before calling `SetOptionBytes`.

Comment (jkralik) [reply message/pool/message.go:141]: Ok. pls create `AddETag`

Comment (Danielius1922) [reply message/pool/message.go:142]: Finally created! :D

**Pull request merged**